examples/dataproc_provisioner_example/main.tf

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.0"
+}
+
+data "google_client_config" "current" {
+}
+
+provider "cdap" {
+    host = "https://adp-ingestion-n-gd-mps-adp-afhiretl-n-d393-dot-usc1.datafusion.googleusercontent.com/api/"
```

Comment:
Replace host with a var? Or put an example host?

Reply:
Would it be safe to use the [`service_endpoint`](https://www.terraform.io/docs/providers/google/r/data_fusion_instance.html#service_endpoint) output of the datafusion instance resource here?

I think it'd be the most natural user experience, however, it is not in line with the guidance about input variables in the [provider docs](https://www.terraform.io/docs/configuration/providers.html#provider-configuration) 
which state: 
>The configuration arguments defined by the provider may be assigned using expressions, which can for example allow them to be parameterized by input variables. However, since provider configurations must be evaluated in order to perform any resource type action, provider configurations may refer only to values that are known before the configuration is applied. In particular, avoid referring to attributes exported by other resources unless their values are specified directly in the configuration.
I think it would behave badly on the first apply.
@umairidris @morgante have you seen a good way around this sort of thing in the past?
Would this be a time where we'd say "create the instance using google provider in one workspace and use the cdap provider resources in another (getting the url from remote state)?"

Reply:
I am not sure if the docs are outdated, but it's definitely possible to use the CDF resource as I have typically done so. In fact the provider main page shows an example of it: https://googlecloudplatform.github.io/terraform-provider-cdap/

And here's another example of the same pattern: https://medium.com/google-cloud/bootstrapping-google-kubernetes-engine-after-creating-it-dca595f830a1

Reply:
In general, it's not necessarily a *recommended* practice though it is frequently done. Unfortunately because it's not officially supported, YMMV and some cases will work while others will break unexpectedly (for example, the Kubernetes provider just broke this: https://github.com/terraform-providers/terraform-provider-kubernetes/issues/759).

The safest option is to split them into two workspaces.

That being said, it does certainly unlock some use cases and since we control the provider there's hopefully less risk of unexpected breakage.